packages/gatsby/index.d.ts

```diff
@@ -11,24 +14,859 @@ export {
+  stage: string;
+}
+
+export interface CreateDevServerArgs extends ParentSpanPluginArgs {
+  app: Application;
+}
+
+export interface CreateNodeArgs extends ParentSpanPluginArgs {
+  node: Node;
+  traceId: string;
+  traceTags: {
+    nodeId: string;
+    nodeType: string;
+  };
+}
+
+export interface CreatePageArgs extends ParentSpanPluginArgs {
+  page: Node;
+  traceId: string;
+}
+
+export interface CreateWebpackConfigArgs extends ParentSpanPluginArgs {
+  getConfig: Function;
+  stage: string;
+  rules: WebpackRules;
+  loaders: WebpackLoaders;
+  plugins: WebpackPlugins;
+}
+
+export interface PreprocessSourceArgs extends ParentSpanPluginArgs {
+  filename: string;
+  contents: string;
+}
+
+export interface ResolvableExtensionsArgs extends ParentSpanPluginArgs {
+  traceId: string;
+}
+
+export interface SetFieldsOnGraphQLNodeTypeArgs extends ParentSpanPluginArgs {
+  type: {
+    name: string;
+    nodes: any[];
+  };
+  traceId: string;
+}
+
+export interface SourceNodesArgs extends ParentSpanPluginArgs {
+  traceId: string;
+  waitForCascadingActions: boolean;
+}
+
+export interface PreRenderHTMLArgs extends NodePluginArgs {
+  getHeadComponents: any[];
+  replaceHeadComponents: Function;
+  getPreBodyComponents: any[];
+  replacePreBodyComponents: Function;
+  getPostBodyComponents: any[];
+  replacePostBodyComponents: Function;
+}
+
+export interface RenderBodyArgs extends NodePluginArgs {
+  pathname: string;
+  setHeadComponents: Function;
+  setHtmlAttributes: Function;
+  setBodyAttributes: Function;
+  setPreBodyComponents: Function;
+  setPostBodyComponents: Function;
+  setBodyProps: Function;
+}
+
+export interface ReplaceRendererArgs extends NodePluginArgs {
+  replaceBodyHTMLString: Function;
+  setHeadComponents: Function;
+  setHtmlAttributes: Function;
+  setBodyAttributes: Function;
+  setPreBodyComponents: Function;
+  setPostBodyComponents: Function;
+  setBodyProps: Function;
+}
+
+export interface WrapPageElementNodeArgs extends NodePluginArgs {
+  element: object;
+  props: object;
+  pathname: string;
+}
+
+export interface WrapRootElementNodeArgs extends NodePluginArgs {
+  element: object;
+}
+
+export interface ParentSpanPluginArgs extends NodePluginArgs {
+  parentSpan: object;
+}
+
+export interface NodePluginArgs {
+  pathPrefix: string;
+  boundActionCreators: Actions;
+  actions: Actions;
+  loadNodeContent: Function;
+  store: Store;
+  emitter: EventEmitter;
+  getNodes: Function;
+  getNode: Function;
+  getNodesByType: Function;
+  hasNodeChanged: Function;
+  reporter: Reporter;
+  getNodeAndSavePathDependency: Function;
+  cache: Cache;
+  createNodeId: Function;
+  createContentDigest: Function;
+  tracing: Tracing;
+  [key: string]: unknown;
+}
+
+export interface Actions {
+  /** @see https://www.gatsbyjs.org/docs/actions/#deletePage */
+  deletePage: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#createPage */
+  createPage(args: CreatePageArgs, option?: ActionOptions): void;
```

Comment:
CreatePageArgs isn't really correct for this function

We'll need to reuse plugin a few types so probably best to create a type for it
```ts
interface ActionPlugin {
  name: string;
}
```

```suggestion
  createPage(args: {path: string, component: string, context: Record<string, unknown>}, plugin?: ActionPlugin, option?: ActionOptions): void;
```